lang/cs/Org.Apache.REEF.Client/Multi/MultiRuntimeConfigurationBuilder.cs

```diff
@@ -0,0 +1,256 @@
+        {
+            _hdiUsername = userName;
+            return this;
+        }
+
+        public MultiRuntimeConfigurationBuilder SetHDInsightPassword(string pwd)
+        {
+            _hdiPassword = pwd;
+            return this;
+        }
+
+        public MultiRuntimeConfigurationBuilder SetHDInsightUrl(string hdiUrl)
+        {
+            _hdiUrl = hdiUrl;
+            return this;
+        }
+
+        public MultiRuntimeConfigurationBuilder SetAzureBlobContainerName(string containerName)
+        {
+            _jobSubmissionDirectoryPrefix = containerName;
+            return this;
+        }
+
+        public MultiRuntimeConfigurationBuilder SetAzureBlobConnectionString(string connectionString)
+        {
+            _connectionString = connectionString;
+            return this;
+        }
+
+        public IConfiguration Build()
+        {
+            if (!this._submissionRuntime.HasValue)
+            {
+                throw new ArgumentException("Submission runtime was not set");
+            }
+
+            if (!_defaultRuntime.IsPresent() && _runtimes.Count == 1)
+            {
+                _defaultRuntime = Optional<RuntimeName?>.Of(_runtimes.First());
+            }
+
+            if (!_defaultRuntime.IsPresent())
+            {
+                throw new ArgumentException("Default runtime was not set");
+            }
+
+            if (_runtimes.Contains(RuntimeName.Local) && !_maxEvaluatorsNumber.HasValue)
+            {
+                throw new ArgumentException("Max evaluators number should be a positive integer");
+            }
+
+            IList<string> runtimes = _runtimes.Select(x => x.ToString()).ToList();
+            var config = TangFactory.GetTang().NewConfigurationBuilder()
+                .BindNamedParameter<DefaultRuntime, string>(GenericType<DefaultRuntime>.Class, _defaultRuntime.Value.ToString());
+
+            foreach (var runtime in runtimes)
+            {
+                config.BindSetEntry<DefinedRuntimes, string>(GenericType<DefinedRuntimes>.Class, runtime);
+            }
+                
+            AvroConfigurationSerializer serializer = new AvroConfigurationSerializer();
+            var serializedConfig = serializer.ToString(config.Build());
+            var configBuilder = TangFactory.GetTang().NewConfigurationBuilder()
+                .BindImplementation(
+                    GenericType<IYarnREEFDotNetAppParamsSerializer>.Class,
+                    GenericType<YarnMultiREEFDotNetAppParamSerializer>.Class)
+                .BindNamedParameter<LauncherClassName, string>(
+                        GenericType<LauncherClassName>.Class,
+                        MultiRuntimeLauncher)
+                .BindNamedParameter<AdditionalDriverAppConfiguration, string>(
+                        GenericType<AdditionalDriverAppConfiguration>.Class,
+                        serializedConfig);
+           
+            switch (this._submissionRuntime.Value)
```

Comment:
I though I did... Did another pass, though we have a lot if files (not part of this PR) that using this.